File: utilities/embedded_samples/sample.go
package main

import (
	"fmt"
	"os"
)

type User struct {
	Name string
}

func (u *User) Greet() {
	fmt.Printf("Hello, %s!\n", u.Name)
}

func main() {
	user := &User{Name: "World"}
	if len(os.Args) > 1 {
		user.Name = os.Args[1]
	}
	user.Greet()
}

File: utilities/embedded_samples/sample.java
// This is a basic Java class
package com.example;

import java.util.List;

public class Greeter {
    private final String name;

    public Greeter(String name) {
        this.name = name;
    }

    public void greet(List<String> others) {
        for (String other : others) {
            System.out.println("Hello, " + other + "! I'm " + name);
        }
    }
}
